tfhe/src/integer/server_key/radix/even_odd.rs

```diff
@@ -63,7 +63,11 @@ impl ServerKey {
     {
         // Even though we don't need to worry about carries,
         // we still clean them
-        if !ct.block_carries_are_empty() {
+        if ct
+            .blocks()
+            .iter()
+            .any(|block| self.key.max_degree.validate(block.degree).is_err())
```

Comment:
any reason for this change ?

Reply:
its related to this comment https://github.com/zama-ai/tfhe-rs/pull/1488#discussion_r1731437886

this function does not care about carries, so a smart variant does no make sense unless there is some carry cleaning (or we accept to put a #[allow] of some sort

Reply:
ah right, I think we have some unused &mut on smart variants where cleaning is not done we end up with an allow IIRC